Fixes #8 - Update code to create an account on __init__ and delete it on __del__

bobsilverberg opened the pull request:
@davehunt r?

davehunt commented on tests/test_fxa_test_account.py:
```diff
@@ -5,26 +5,42 @@
 
+    @pytest.mark.skip_selenium
     def test_new_account_pw_does_not_contain_numbers(self, dev_account):
         assert re.search(r'\d', dev_account.password) is None
```
Unrelated to your changes, but why do we only run this on dev?

bobsilverberg replied:
It's because it's only really needed on one evn, as there is only one path for the code that is executed to set the password regardless of the env. I suppose there is no reason that we shouldn't also run it against prod, other than the fact that it's unnecessary, but that is based on the current implementation. Although I cannot imagine that part of the code changing, I suppose it's possible at some point that different envs might encounter different code paths, so to future-proof against coverage loss we should probably run in on both. I'll update that.

davehunt commented on tests/test_fxa_test_account.py:
```diff
@@ -5,26 +5,42 @@
+        """ Check that the __del__ method does destroy the FxA """
+        account = FxATestAccount()
+        saved_account_info = {'email': account.email,
+                              'password': account.password}
```
Nit: I don't think we need a dict here, we can just use variables named email and password.

bobsilverberg replied:
Good point

davehunt commented:
Looks good, and turns out to be a pretty simple change. Are you pleased with it?

bobsilverberg commented:
Thanks for the review @davehunt. I addressed your comments, plus I added logging to the `__del__` method to match that in place for `__init__`.

Yes, I am pleased with the new version. Did it address all of your initial concerns about design, or do you think it needs more changes?

davehunt commented:
With README updated, r+

bobsilverberg commented:
This is ready to go. I'm going to merge and release it.
